src/toil/jobStores/aws/jobStore.py

```diff
@@ -149,22 +151,44 @@ class AWSJobStore(AbstractJobStore):
     partitioned into chunks of 1024 bytes and each chunk is stored as a an attribute of the SDB
     item representing the job. UUIDs are used to identify jobs and files.
     """
+    class Locator(namedtuple('Locator', ('region', 'namePrefix')), AbstractJobStore.Locator):
+        """
+        Parses a job store locator string into an AWS locator object with the following attributes:
+            * region: The AWS region of the job store as a string.
+            * namePrefix: The name prefix of the job store as a string.
 
-    @classmethod
-    def loadOrCreateJobStore(cls, locator, config=None, **kwargs):
-        region, namePrefix = locator.split(':')
-        if not cls.bucketNameRe.match(namePrefix):
-            raise ValueError("Invalid name prefix '%s'. Name prefixes must contain only digits, "
-                             "hyphens or lower-case letters and must not start or end in a "
-                             "hyphen." % namePrefix)
-        # Reserve 13 for separator and suffix
-        if len(namePrefix) > cls.maxBucketNameLen - cls.maxNameLen - len(cls.nameSeparator):
-            raise ValueError("Invalid name prefix '%s'. Name prefixes may not be longer than 50 "
-                             "characters." % namePrefix)
-        if '--' in namePrefix:
-            raise ValueError("Invalid name prefix '%s'. Name prefixes may not contain "
-                             "%s." % (namePrefix, cls.nameSeparator))
-        return cls(region, namePrefix, config=config, **kwargs)
+        The syntax for an AWS job store locator string is as follows:
+            aws:<AWS region>:<name prefix>
+
+        For name prefix syntax see exception messages in parse method below.
+        """
+        @property
+        def outer(self):
```

Comment:
Here's the main quirk I mentioned at stand up. The method we discussed 
```
class Locator(...):
    outer = AWSJobStore
   ...
```
does not work.